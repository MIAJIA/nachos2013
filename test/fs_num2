Purpose:
Test to create many directories in root.

Usage:
-- nachos.ag.FilesysGrader -x fs_num2.coff -# output=fs_num2.out,coffPar0=XXX

Author:
Meilun Sheng

History:
2010-01-07 Meilun Sheng: The first version.
2010-01-07 Bo Tang: The second version. Add coffPar0

Arguments:
-output fs_num2.out // to compare answer
-coffPar0=XXX // the number of directories autograder will create (less than 500)

Sample:
-- nachos.ag.FilesysGrader -x fs_num2.coff -# output=fs_num2.out,coffPar0=100

Restrictions:
To run this test, you should follow the specification in filesys_spec
ATTENTION!!! Your directory must hold XXX files at least to pass it.
If you copy other files into your disk, make your directory more powerful.
ATTENTION2!!! This tester should be really slow, so be patient, or take a cup of tea. 