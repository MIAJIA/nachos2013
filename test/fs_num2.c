#include "filesysgrader.h"
#include "syscall.h"
#include "stdio.h"
#include "stdlib.h"

#define BUFSIZE 1024
#define BUFSIZE_s 50
char loc[BUFSIZE+1];
char loc2[BUFSIZE+1];
char buf[BUFSIZE+1];
char buf2[BUFSIZE_s];
int main(int argc, char** argv)
{
// if your file system doesn't support more than count files per dictory
// you must be failed
  printf("tester is surly slow, be patient.\n");
  int i=0;
  int count = readParameter(0);
// phase 1 directory create
  for(i=0;i<count;i++)
  {
    sprintf(loc2,"%d\0",i);
    printf("Creating %s\n",loc2);
    assertTrueWMsg(mkdir(loc2)==0,"Create failed");		
  }
  assertTrueWMsg(chdir(loc2)==0,"Create error");
  assertTrueWMsg(chdir("/1")==0,"Create error 2");
  done();
  return 0;
}
