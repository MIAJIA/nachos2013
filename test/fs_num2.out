tester is surly slow, be patient.
Creating 0
Creating 1
Creating 2
Creating 3
Creating 4
Creating 5
Creating 6
Creating 7
Creating 8
Creating 9
Creating 10
Creating 11
Creating 12
Creating 13
Creating 14
Creating 15
Creating 16
Creating 17
Creating 18
Creating 19
Creating 20
Creating 21
Creating 22
Creating 23
Creating 24
Creating 25
Creating 26
Creating 27
Creating 28
Creating 29
Creating 30
Creating 31
Creating 32
Creating 33
Creating 34
Creating 35
Creating 36
Creating 37
Creating 38
Creating 39
Creating 40
Creating 41
Creating 42
Creating 43
Creating 44
Creating 45
Creating 46
Creating 47
Creating 48
Creating 49
Creating 50
Creating 51
Creating 52
Creating 53
Creating 54
Creating 55
Creating 56
Creating 57
Creating 58
Creating 59
Creating 60
Creating 61
Creating 62
Creating 63
Creating 64
Creating 65
Creating 66
Creating 67
Creating 68
Creating 69
Creating 70
Creating 71
Creating 72
Creating 73
Creating 74
Creating 75
Creating 76
Creating 77
Creating 78
Creating 79
Creating 80
Creating 81
Creating 82
Creating 83
Creating 84
Creating 85
Creating 86
Creating 87
Creating 88
Creating 89
Creating 90
Creating 91
Creating 92
Creating 93
Creating 94
Creating 95
Creating 96
Creating 97
Creating 98
Creating 99
Creating 100
Creating 101
Creating 102
Creating 103
Creating 104
Creating 105
Creating 106
Creating 107
Creating 108
Creating 109
Creating 110
Creating 111
Creating 112
Creating 113
Creating 114
Creating 115
Creating 116
Creating 117
Creating 118
Creating 119
Creating 120
Creating 121
Creating 122
Creating 123
Creating 124
Creating 125
Creating 126
Creating 127
Creating 128
Creating 129
Creating 130
Creating 131
Creating 132
Creating 133
Creating 134
Creating 135
Creating 136
Creating 137
Creating 138
Creating 139
Creating 140
Creating 141
Creating 142
Creating 143
Creating 144
Creating 145
Creating 146
Creating 147
Creating 148
Creating 149
Creating 150
Creating 151
Creating 152
Creating 153
Creating 154
Creating 155
Creating 156
Creating 157
Creating 158
Creating 159
Creating 160
Creating 161
Creating 162
Creating 163
Creating 164
Creating 165
Creating 166
Creating 167
Creating 168
Creating 169
Creating 170
Creating 171
Creating 172
Creating 173
Creating 174
Creating 175
Creating 176
Creating 177
Creating 178
Creating 179
Creating 180
Creating 181
Creating 182
Creating 183
Creating 184
Creating 185
Creating 186
Creating 187
Creating 188
Creating 189
Creating 190
Creating 191
Creating 192
Creating 193
Creating 194
Creating 195
Creating 196
Creating 197
Creating 198
Creating 199
Creating 200
Creating 201
Creating 202
Creating 203
Creating 204
Creating 205
Creating 206
Creating 207
Creating 208
Creating 209
Creating 210
Creating 211
Creating 212
Creating 213
Creating 214
Creating 215
Creating 216
Creating 217
Creating 218
Creating 219
Creating 220
Creating 221
Creating 222
Creating 223
Creating 224
Creating 225
Creating 226
Creating 227
Creating 228
Creating 229
Creating 230
Creating 231
Creating 232
Creating 233
Creating 234
Creating 235
Creating 236
Creating 237
Creating 238
Creating 239
Creating 240
Creating 241
Creating 242
Creating 243
Creating 244
Creating 245
Creating 246
Creating 247
Creating 248
Creating 249
Creating 250
Creating 251
Creating 252
Creating 253
Creating 254
Creating 255
Creating 256
Creating 257
Creating 258
Creating 259
Creating 260
Creating 261
Creating 262
Creating 263
Creating 264
Creating 265
Creating 266
Creating 267
Creating 268
Creating 269
Creating 270
Creating 271
Creating 272
Creating 273
Creating 274
Creating 275
Creating 276
Creating 277
Creating 278
Creating 279
Creating 280
Creating 281
Creating 282
Creating 283
Creating 284
Creating 285
Creating 286
Creating 287
Creating 288
Creating 289
Creating 290
Creating 291
Creating 292
Creating 293
Creating 294
Creating 295
Creating 296
Creating 297
Creating 298
Creating 299
Creating 300
Creating 301
Creating 302
Creating 303
Creating 304
Creating 305
Creating 306
Creating 307
Creating 308
Creating 309
Creating 310
Creating 311
Creating 312
Creating 313
Creating 314
Creating 315
Creating 316
Creating 317
Creating 318
Creating 319
Creating 320
Creating 321
Creating 322
Creating 323
Creating 324
Creating 325
Creating 326
Creating 327
Creating 328
Creating 329
Creating 330
Creating 331
Creating 332
Creating 333
Creating 334
Creating 335
Creating 336
Creating 337
Creating 338
Creating 339
Creating 340
Creating 341
Creating 342
Creating 343
Creating 344
Creating 345
Creating 346
Creating 347
Creating 348
Creating 349
Creating 350
Creating 351
Creating 352
Creating 353
Creating 354
Creating 355
Creating 356
Creating 357
Creating 358
Creating 359
Creating 360
Creating 361
Creating 362
Creating 363
Creating 364
Creating 365
Creating 366
Creating 367
Creating 368
Creating 369
Creating 370
Creating 371
Creating 372
Creating 373
Creating 374
Creating 375
Creating 376
Creating 377
Creating 378
Creating 379
Creating 380
Creating 381
Creating 382
Creating 383
Creating 384
Creating 385
Creating 386
Creating 387
Creating 388
Creating 389
Creating 390
Creating 391
Creating 392
Creating 393
Creating 394
Creating 395
Creating 396
Creating 397
Creating 398
Creating 399
Creating 400
Creating 401
Creating 402
Creating 403
Creating 404
Creating 405
Creating 406
Creating 407
Creating 408
Creating 409
Creating 410
Creating 411
Creating 412
Creating 413
Creating 414
Creating 415
Creating 416
Creating 417
Creating 418
Creating 419
Creating 420
Creating 421
Creating 422
Creating 423
Creating 424
Creating 425
Creating 426
Creating 427
Creating 428
Creating 429
Creating 430
Creating 431
Creating 432
Creating 433
Creating 434
Creating 435
Creating 436
Creating 437
Creating 438
Creating 439
Creating 440
Creating 441
Creating 442
Creating 443
Creating 444
Creating 445
Creating 446
Creating 447
Creating 448
Creating 449
Creating 450
Creating 451
Creating 452
Creating 453
Creating 454
Creating 455
Creating 456
Creating 457
Creating 458
Creating 459
Creating 460
Creating 461
Creating 462
Creating 463
Creating 464
Creating 465
Creating 466
Creating 467
Creating 468
Creating 469
Creating 470
Creating 471
Creating 472
Creating 473
Creating 474
Creating 475
Creating 476
Creating 477
Creating 478
Creating 479
Creating 480
Creating 481
Creating 482
Creating 483
Creating 484
Creating 485
Creating 486
Creating 487
Creating 488
Creating 489
Creating 490
Creating 491
Creating 492
Creating 493
Creating 494
Creating 495
Creating 496
Creating 497
Creating 498
Creating 499
Creating 500
